sdgym/synthesizers/uniform.py

```diff
@@ -61,12 +62,72 @@ def _sample_from_synthesizer(self, synthesizer, n_samples):
         for name, column in transformed.items():
             kind = column.dtype.kind
             if kind == 'i':
-                values = np.random.randint(column.min(), column.max() + 1, size=self.length)
+                values = np.random.randint(
+                    int(column.min()), int(column.max()) + 1, size=n_samples, dtype=np.int64
+                )
             elif kind in ['O', 'b']:
-                values = np.random.choice(column.unique(), size=self.length)
+                values = np.random.choice(column.unique(), size=n_samples)
             else:
-                values = np.random.uniform(column.min(), column.max(), size=self.length)
-
+                values = np.random.uniform(column.min(), column.max(), size=n_samples)
             sampled[name] = values
 
         return hyper_transformer.reverse_transform(sampled)
+
+
+class MultiTableUniformSynthesizer(MultiTableBaselineSynthesizer):
+    """Multi-table Uniform Synthesizer.
+
+    This synthesizer trains a UniformSynthesizer on each table in the multi-table dataset.
+    It samples data from each table independently using the corresponding trained synthesizer.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.num_rows_per_table = {}
+
+    def _get_trained_synthesizer(self, data, metadata):
+        """Train a UniformSynthesizer for each table in the multi-table dataset.
+
+        Args:
+            data (dict):
+                A dict mapping table name to table data.
+            metadata (sdv.metadata.Metadata):
+                The metadata
+
+        Returns:
+            A dict mapping table name to trained UniformSynthesizer instance.
+        """
+        synthesizers = {}
+        for table_name, table_data in data.items():
+            self.num_rows_per_table[table_name] = len(table_data)
+            table_metadata = metadata.get_table_metadata(table_name)
+            synthesizer = UniformSynthesizer()
+            trained_synthesizer = synthesizer._get_trained_synthesizer(table_data, table_metadata)
+            synthesizers[table_name] = trained_synthesizer
+
+        return synthesizers
```

Comment:
why don't we follow the pattern else where and return an instance of MultiTableUniformSynthesizer? I think _get_trained_synthesizer should be a static or class method also since it returns an instance in other places like here
https://github.com/sdv-dev/SDGym/blob/2a42a24cb9e86ad11042281c587389dad8ed02f6/sdgym/synthesizers/sdv.py#L45-L47